comp/core/workloadmeta/collectors/internal/containerd/container_builder.go

```diff
@@ -154,3 +161,51 @@ func extractRuntimeFlavor(runtime string) workloadmeta.ContainerRuntimeFlavor {
 	}
 	return workloadmeta.ContainerRuntimeFlavorDefault
 }
+
+// extractRepoDigestFromImage extracts the repoDigest from workloadmeta store if it exists and unique
+// the format of repoDigest is "registry/repo@sha256:digest", the format of return value is "sha256:digest"
+func extractRepoDigestFromImage(imageID string, imageRegistry string, store workloadmeta.Component) string {
+	existingImgMetadata, err := store.GetImage(imageID)
```

Comment:
Would this create a different race condition inside of `notifyInitialEvents`? We are collecting image metadata before we are building containers, but we have no guarantee that the image metadata will be in the workloadmeta store by the time we hit this line, since it is running in a different goroutine.

Since we are already talking to the containerd client several times in `buildWorkloadMetaContainer`, does it make sense to get the digest from the containerd client directly, in a similar manner to [how we get the data to construct the image metadata event](https://github.com/DataDog/datadog-agent/blob/e219615a01053e9090644b55ad88e754cf39abac/comp/core/workloadmeta/collectors/internal/containerd/image.go#L251-L257)?

Reply:
Yeah, the original plan was to reuse containerd client directly. However, there is another ongoing pr fix not merged yet: [Consolidate multiple image references in initial containerd image collection](https://github.com/DataDog/datadog-agent/pull/22355/files).  I need this pr merged first because 
1. it fixed repo-digest missing in some references returned by containerd client
2. I can then reuse this [function](https://github.com/DataDog/datadog-agent/pull/22355/files#diff-80f83c7f00050ea3349894f491fc122e69fd17325d1688ca3ede4a60754dd24aR267)